src/Features/Core/Portable/InitializeParameter/AbstractAddParameterCheckCodeRefactoringProvider.cs

```diff
@@ -237,8 +237,15 @@ private static bool IsIfNullCheck(IOperation statement, IParameterSymbol paramet
         protected bool ParameterValidForNullCheck(Document document, IParameterSymbol parameter, SemanticModel semanticModel,
             IBlockOperation blockStatementOpt, CancellationToken cancellationToken)
         {
-            if (!parameter.Type.IsReferenceType &&
-                !parameter.Type.IsNullable())
+            if (parameter.Type.IsReferenceType)
+            {
+                // Don't add null checks to things explicitly declared nullable
+                if (parameter.Type.NullableAnnotation == NullableAnnotation.Annotated)
+                {
+                    return false;
+                }
+            }
+            else if (!parameter.Type.IsNullable())
```

Comment:
```suggestion
            // Don't add null checks to things explicitly declared nullable
            if (parameter.Type is { IsReferenceType: true, NullableAnnotation: NullableAnnotation.Annotated })
                return false;

            if (!parameter.Type.IsNullable())
```

Reply:
@CyrusNajmabadi That will change behavior -- IsNullable is specifically "is this Nullable<T>".